### PR TITLE
Update from boto2 to boto3.

### LINKED ISSUE #8170: Update to Boto3
With the rollout of Python 3.12 on GitHub Actions, it is time for us to migrate to Boto3

Migration guide:
https://boto3.amazonaws.com/v1/documentation/api/latest/guide/migration.html

Repo:
https://github.com/boto/boto3#boto3---the-aws-sdk-for-python

### PR DESCRIPTION
**Pull request by ekharkunov**
Fixes #8170.

### Technical changes
Update to boto3 1.34.69.
Add jsons with services and endpoints described aws services because boto cannot find definition from .whl package.

Do not use urlib3 2.2.0 because of issue https://github.com/boto/botocore/issues/3111
Fixed in https://github.com/urllib3/urllib3/releases/tag/2.2.1

## PR checklist

* [ ] Code
	* [ ] Add engine and/or editor unit tests.
	* [ ] New and changed code follows the overall code style of existing code
	* [ ] Add comments where needed
* [ ] Documentation
	* [ ] Make sure that API documentation is updated in code comments
	* [ ] Make sure that manuals are updated (in github.com/defold/doc)
* [ ] Prepare pull request and affected issue for automatic release notes generator
	* [ ] Pull request - Write a message that explains what this pull request does. What was the problem? How was it solved? What are the changes to APIs or the new APIs introduced? This message will be used in the generated release notes. Make sure it is well written and understandable for a user of Defold.
	* [ ] Pull request - Write a pull request title that in a sentence summarises what the pull request does. Do not include "Issue-1234 ..." in the title. This text will be used in the generated release notes.
	* [ ] Pull request - Link the pull request to the issue(s) it is closing. Use on of the [approved closing keywords](https://docs.github.com/en/issues/tracking-your-work-with-issues/linking-a-pull-request-to-an-issue).
	* [ ] Affected issue - Assign the issue to a project. Do not assign the pull request to a project if there is an issue which the pull request closes.
	* [ ] Affected issue - Assign the "breaking change" label to the issue if introducing a breaking change.
	* [ ] Affected issue - Assign the "skip release notes" is the issue should not be included in the generated release notes.

----------

Example of a well written PR description:

1. Start with the user facing changes. This will end up in the release notes.
1. Add one of the GitHub approved closing keywords
1. Optionally also add the technical changes made. This is information that might help the reviewer. It will not show up in the release notes. Technical changes are identified by a line starting with one of these:
   1. `### Technical changes` 
   1. `Technical changes:`
   2. `Technical notes:`

```
There was a anomaly in the carbon chroniton propeller, introduced in version 8.10.2. This fix will make sure to reset the phaser collector on application startup.

Fixes #1234

### Technical changes
* Pay special attention to line 23 of phaser_collector.clj as it contains some interesting optimizations
* The propeller code was not taking into account a negative phase.
```

### CODE REVIEW

File: scripts/move_s3_release.py [hunk 1]
```diff
@@ -24,8 +24,14 @@
...
 if __name__ == '__main__':
-    boto_path = os.path.normpath(os.path.join(os.path.dirname(os.path.abspath(__file__)), '../packages/boto-2.28.0-py2.7.egg'))
+    urllib_path = os.path.normpath(os.path.join(os.path.dirname(os.path.abspath(__file__)), '../packages/urllib3-2.2.1-py3-none-any.whl'))
+    botocore_path = os.path.normpath(os.path.join(os.path.dirname(os.path.abspath(__file__)), '../packages/botocore-1.34.69-py3-none-any.whl'))
+    boto_path = os.path.normpath(os.path.join(os.path.dirname(os.path.abspath(__file__)), '../packages/boto3-1.34.69-py3-none-any.whl'))
+    data_path = os.path.normpath(os.path.join(os.path.dirname(os.path.abspath(__file__)), '../packages/boto3_data'))
+    sys.path.insert(0, urllib_path)
+    sys.path.insert(0, botocore_path)
     sys.path.insert(0, boto_path)
+    os.environ['AWS_DATA_PATH'] = data_path
```
Comment (JCash) [inline scripts/move_s3_release.py:34]: Could this be moved into one place, like `s3.init_boto()` or similar.
It's easy to miss this file.

File: scripts/build.py [hunk 2]
```diff
@@ -2195,8 +2209,15 @@ def _form_env(self):
...
 if __name__ == '__main__':
-    boto_path = os.path.normpath(os.path.join(os.path.dirname(os.path.abspath(__file__)), '../packages/boto-2.28.0-py2.7.whl'))
+    urllib_path = os.path.normpath(os.path.join(os.path.dirname(os.path.abspath(__file__)), '../packages/urllib3-2.2.1-py3-none-any.whl'))
+    botocore_path = os.path.normpath(os.path.join(os.path.dirname(os.path.abspath(__file__)), '../packages/botocore-1.34.69-py3-none-any.whl'))
+    boto_path = os.path.normpath(os.path.join(os.path.dirname(os.path.abspath(__file__)), '../packages/boto3-1.34.69-py3-none-any.whl'))
+    data_path = os.path.normpath(os.path.join(os.path.dirname(os.path.abspath(__file__)), '../packages/boto3_data'))
+    sys.path.insert(0, urllib_path)
+    sys.path.insert(0, botocore_path)
     sys.path.insert(0, boto_path)
+    os.environ['AWS_DATA_PATH'] = data_path
```
Comment (JCash) [inline scripts/build.py:2219]: Not sure if we should fix it now (old issue), but it feels strange that we install the .whl's (when running `install_ext`), yet still do this manual touch up.

Ideally, we should only need s3, when we run those commands.

**Review by JCash — DISMISSED** [review]
Looks good to me.
Some minor comments that aren't crucial.

File: scripts/build.py [hunk 3]
```diff
@@ -2195,8 +2209,15 @@ def _form_env(self):
         return env
 
 if __name__ == '__main__':
-    boto_path = os.path.normpath(os.path.join(os.path.dirname(os.path.abspath(__file__)), '../packages/boto-2.28.0-py2.7.whl'))
+    urllib_path = os.path.normpath(os.path.join(os.path.dirname(os.path.abspath(__file__)), '../packages/urllib3-2.2.1-py3-none-any.whl'))
```
[the remark this answers is not in the document]
Comment (ekharkunov) [reply scripts/build.py]: Removed. Seems it works.

File: scripts/move_s3_release.py (hunk 1 above)
Comment (ekharkunov) [reply scripts/move_s3_release.py]: Create `s3.init_boto_data_path` function.

File: scripts/build.py (hunk 2 above)
Comment (ekharkunov) [reply scripts/build.py]: Removed.

File: scripts/build.py [hunk 4]
```diff
@@ -2209,15 +2209,7 @@ def _form_env(self):
...
-    urllib_path = os.path.normpath(os.path.join(os.path.dirname(os.path.abspath(__file__)), '../packages/urllib3-2.2.1-py3-none-any.whl'))
-    botocore_path = os.path.normpath(os.path.join(os.path.dirname(os.path.abspath(__file__)), '../packages/botocore-1.34.69-py3-none-any.whl'))
-    boto_path = os.path.normpath(os.path.join(os.path.dirname(os.path.abspath(__file__)), '../packages/boto3-1.34.69-py3-none-any.whl'))
-    data_path = os.path.normpath(os.path.join(os.path.dirname(os.path.abspath(__file__)), '../packages/boto3_data'))
-    sys.path.insert(0, urllib_path)
-    sys.path.insert(0, botocore_path)
-    sys.path.insert(0, boto_path)
-    os.environ['AWS_DATA_PATH'] = data_path
-
+    s3.init_boto_data_path()
```
Comment (JCash) [inline scripts/build.py:2212]: ❤️

**Review by JCash — APPROVED** [review]
lgtm!

**Pull request merged**